src/switcher-ui.tsx

```
@@ -656,18 +666,48 @@ function SwitcherApp() {
   };
+  // Persist the two header toggles from effects, never from inside a state
+  // updater: an updater must be pure (React is free to call it more than once),
+  // and one effect owning each key means no second writer can disagree with it
+  // about what is stored — the reset below just sets state and this follows.
+  useEffect(() => {
+    try {
+      localStorage.setItem(
+        'codev-pinned-collapsed',
+        pinnedCollapsed ? '1' : '0',
+      );
+    } catch {}
```

ESLint

[error] Empty block statement.

(no-empty)